pkg/daemon/ceph/client/filesystem.go

```diff
@@ -188,36 +181,38 @@ func SetNumMDSRanks(context *clusterd.Context, clusterName, fsName string, activ
 			fsName, activeMDSCount, err)
 	}
 
-	// ** Noted section 2 - See note at top of function
-	// Now check the error to see if we can even determine whether we should reduce or not
-	if errAtStart != nil {
-		return fmt.Errorf(`failed to get filesystem %s info needed to ensure mds rank can be changed correctly,
-if Ceph version is Luminous (12.y.z) and num active mdses (max_mds) was lowered, USER should deactivate extra active mdses manually: %v`,
-			fsName, errAtStart)
-	}
-	if int(activeMDSCount) > fsAtStart.MDSMap.MaxMDS {
-		return nil // No need to deactivate mdses if we are raising max_mds
-	}
-	logger.Debugf("deactivating some running mdses for filesystem %s", fsName)
-	// Deactivate all mdses except desired number (N); arbitrarily choose first N to live
-	fs, err := GetFilesystem(context, clusterName, fsName)
-	if err != nil {
-		logger.Warningf(
-			fmt.Sprintf("Failed to get filesystem %s info needed to deactivate running mdses. ", fsName) +
-				"using slightly stale info, this could (rarely) result in momentary loss of filesystem availability" +
-				fmt.Sprintf(": %v", err),
-		)
-		fs = fsAtStart // <-- Do the best we can to disable mdses that were active when we started
-		// Effects of stale info should be non-destructive, & unlikely the info is actually bad
-	}
-	// Deactivate any mdses with a higher rank than the desired max rank
-	// Ceph only allows mdses to be deactivated in reverse order starting with the highest rank
-	for gid := int(len(fs.MDSMap.In)) - 1; gid >= int(activeMDSCount); gid-- {
-		if err := deactivateMdsWithRetry(context, gid, clusterName, fsName); err != nil {
-			logger.Warningf("in mimic+ this error means nothing. in luminous this is non-ideal but not necessarily critical: %v", err)
+	// In Ceph versions earlier than Mimic, need to individually deactivate active MDSes to scale down
+	// the number of active MDSes.
+	if !cephv1.VersionAtLeast(cephVersionName, cephv1.Mimic) {
```

Comment:
Oh I see. What already existed is now enclosed in an if statement; I missed that before.

Go methodologies recommend using fail-fast type statements and eliminating large conditional blocks where possible. I think that this function will read better if this is changed to 
```go
if cephv1.VersionAtLeast(cephVersionName, cephv1.Mimic) {
    return nil
}
// all of the original code
```